packages/components/organisms/f-form/src/components/FormField.vue

```diff
@@ -0,0 +1,150 @@
+
+export default {
+    components: {
+        ErrorMessage,
+        FormField
+    },
+
+    props: {
+        fieldData: {
+            type: Object,
+            required: true
+        }
+    },
+
+    inject: ['$v'],
+
+    computed: {
+        kebabCase () {
+            return this.fieldData.name.replace(/([a-z0-9]|(?=[A-Z]))([A-Z1-9])/g, '$1-$2').toLowerCase();
+        },
+
+        translations () {
+            return {
+                ...this.fieldData.translations,
+                name: `${this.kebabCase}`,
+                errorName: `${this.kebabCase}-error`,
+                emptyError: `error-${this.kebabCase}-empty`,
+                invalidError: `error-${this.kebabCase}-invalid`
+            };
+        },
+
+        fieldProps () {
+            let props = {
+                name: this.translations.name,
+                value: this.fieldData.value,
+                'label-text': this.translations.label,
+                'input-type': this.inputType,
+                'max-length': this.fieldData.mexLength || null
+            };
+
+            if (this.ariaLabel) {
+                props = {
+                    ...props,
+                    'aria-label': this.ariaLabel
+                };
+            }
+
+            if (this.hasError) {
+                props = {
+                    ...props,
+                    'has-error': true,
+                    'aria-invalid': true,
+                    'aria-describedby': this.translations.validationMessages[this.errorType]
+                };
+            }
+
+            return props;
+        },
+
+        inputType () {
+            const inputTypes = {
+                mobileNumber: 'tel',
+                email: 'email',
+                default: 'text'
+            };
+
+            return inputTypes[this.fieldData.name] || inputTypes.default;
+        },
+
+        hasValidationMessages () {
+            return !!this.translations?.validationMessages;
+        },
+
+        hasInvalidErrorMessage () {
+            return !!this.translations?.validationMessages?.invalid;
+        },
+
+        hasError () {
+            return !!this.hasValidationMessages && (this.hasRequiredError || this.hasInvalidError);
+        },
+
+        isEmpty () {
+            return this.validations?.$dirty;
+        },
+
+        validations () {
+            return this.$v.formFields[this.fieldData.name];
+        },
+
+        hasRequiredError () {
+            return this.isEmpty && !this.validations.required;
+        },
+
+        hasInvalidError () {
+            return !!this.hasInvalidErrorMessage && this.isEmpty && !this.validations[this.fieldData.name];
+        },
+
+        errorType () {
+            if (this.hasError) {
+                return this.hasRequiredError ? 'required' : 'invalid';
+            }
+
+            return null;
+        },
+
+        ariaLabel () {
+            if (this.fieldData.name === 'mobileNumber' && this.fieldData.value) {
```

Comment:
Could this also be applied to postcodes, or do we not need to do that at the moment?

Reply:
I just tested it on voiceover and postcodes seem to read fine so I think we're ok to leave it for now

Reply:
Out of interest which postcodes did you try? BS1 might be pronounced fine but would BA1?

Reply:
I've just tried it on BA1 and it still seems fine
Happy to add it as a precaution though?

Reply:
If it works as-is then I'm happy for you to leave it 👍🏻